Updated Main.kt

MarcoChrist opened the pull request:
@nicolaiparlog This PR includes the updated Main.kt with the traffic.csv file, the FileAccess.kt so that reader and writer work properly.
The results.csv which shows whether the cars are able to drive due to trafficjam is also included.

If im not mistaken, this should solve the Issues #28 #29 #30 

Also the other two PR created by NikMass should work fine when all this comes together.

MarcoChrist committed:
Updated Main.kt 

This PR includes the updated Main.kt with the traffic.csv file, the FileAccess.kt so that reader and writer work properly.
The results.csv which shows whether the cars are able to drive due to trafficjam is also included.

nipafx commented on src/main/Resources/traffic.csv:
```diff
@@ -0,0 +1,13 @@
+Car, WantsToDrive
```
The folder should be called `resources`, not `Resources` (capitalization matters).

MarcoChrist replied:
@nicolaiparlog why has this to be lower case? For linux it matters, but why should it be all lower case?

nipafx replied:
I don't know whether there was a particular reason to use lower case back in the old days. I just know it as a convention. A consequence of that is that Maven uses certain paths (like `src/main/resources`) by default and you'd have to actually configure it to use different values.

nipafx commented on src/main/kotlin/FinalProject/Main.kt:
```diff
@@ -1,5 +1,93 @@
-    println("Hello, World!")
+	val Cars : MutableList<Car> = parseCSV()
+
+	scenario(Cars)
```
The `scenario` function is meant to execute an entire simulation (i.e. create cars, simulate, write results) without any CSV input/output. The reason is that you should always have one way to run the simulation _without_ having to rely on CSV parsing (which is easy to break accidentally).

So please change `scenario` to create its own cars and write a second function (maybe `csvScenario`) that loads from CSV, simulates, and writes to CSV. `main` should then execute one or both of these functions.

MarcoChrist replied:
@nicolaiparlog 
The scenario function has a parameter which takes the car list, that means the scenario function itself is independent of CSV. The main function handles the CSV reading/writing

nipafx commented on src/main/kotlin/FinalProject/Main.kt:
```diff
@@ -1,5 +1,93 @@
+		val WantDriveStr : String = record.values[1]
+		val WantDrive : Boolean = WantDriveStr.toBoolean()
+
+		CarListCSV.add(Car(d = WantDrive))
```
Aren't you going to use the `CarId`?

nipafx commented on src/main/kotlin/FinalProject/Main.kt:
```diff
@@ -1,5 +1,93 @@
+
+
+fun parseCSV(): MutableList<Car> {
+	val CarListCSV: MutableList<Car> = mutableListOf()
```
Please start all variable names with lower letters.

nipafx reviewed (changes requested):
Overall this looks very good. But Travis is complaining and you have to fix that.

:bulb: Before any commit you hould make sure that `mvn clean install` does not result in errors.

:bulb: Smaller pull requests that address less issues make it easier to review them.

MarcoChrist committed:
Updated the Main.kt

@nicolaiparlog
The scneario function is now independent from CSV and the static list.

MarcoChrist committed:
Update Main.kt

Changed variable names to lower case.

MarcoChrist committed:
Merge pull request #45 from SJ1337/MarcoChrist-patch-2

Updated the Main.kt
